src/core/credentials/transport/xds/xds_credentials.cc

```diff
@@ -146,7 +146,9 @@ XdsCredentials::create_security_connector(
       auto tls_credentials_options =
           MakeRefCounted<grpc_tls_credentials_options>();
       if (watch_root || watch_identity) {
-        tls_credentials_options->set_certificate_provider(
+        tls_credentials_options->set_root_certificate_provider(
```

Comment:
Please add the following TODO in src/core/xds/grpc/xds_certificate_provider.h:

TODO(roth): Now that we've changed the TLS creds API to configure different providers for root and identity certs, we no longer need to multiplex multiple providers in an XdsCertificateProvider.  Consider removing this code and instead just passing down the relevant TLS creds configuration via a channel arg.